### DISCUSSION TITLE
Example GCN implementation seems to be different from original paper's implementation

### DISCUSSION THREAD

**Discussion by sepiabrown** [snap-stanford/ogb#403] [PCQM4M-LSC]
Looking at the GCN implementation of [here](https://github.com/snap-stanford/ogb/blob/master/examples/lsc/pcqm4m-v2/conv.py) and [here](https://github.com/dmlc/dgl/tree/master/examples/pytorch/ogb_lsc/PCQM4M), they don't seem to follow the implementation on [Semi-supervised classification with graph convolutional networks.](https://arxiv.org/pdf/1609.02907.pdf).

The paper's [implementation](https://github.com/tkipf/gcn/blob/master/gcn/layers.py) seems to put ReLU activation after adding all the convolved variables and bias. But the implementation provided by OGB seems to apply ReLU activation first and then sum up the variables. Also, there is no reference to the idea of adding edge embeddings through `BondEncoder`.

Is the code just a variant of the paper's implementation where ReLU is applied after? Any reference would be appreciated!

**Comment by sepiabrown** [comment]
[@weihua916 ](https://github.com/snap-stanford/ogb/blob/553dc0b70a7a3f95bb47b8e09a1e06ae0aff1e4f/examples/lsc/pcqm4m-v2/conv.py#L60-L64)

**Reply by weihua916** [reply]
Good point. You can consider my implementation as a GCN variant for edge features. I think it is fair to say it is still GCN as long as we use the normalized adjacency matrix and ReLU activation.
